src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -401,6 +405,79 @@ protected override void Dispose(bool disposing)
         base.Dispose(disposing);
     }
 
+    private static Sampler ResolveSamplerFromSpecificationConfigurationKeys(IConfiguration configuration, Sampler stateSampler)
+    {
+        Sampler? sampler = null;
+
+        if (stateSampler != null)
+        {
+            sampler = stateSampler;
+        }
+
+        if (configuration.TryGetStringValue(TracesSamplerConfigKey, out var tracesSampler))
+        {
+            if (sampler != null)
+            {
+                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent(
+                    $"Trace sampler configuration value '{tracesSampler}' has been ignored because a value '{sampler.Description}' was set previously.");
+                return sampler;
+            }
+
+            if (string.Equals(tracesSampler, "always_on", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new AlwaysOnSampler();
+            }
+            else if (string.Equals(tracesSampler, "always_off", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new AlwaysOffSampler();
+            }
+            else if (string.Equals(tracesSampler, "traceidratio", StringComparison.OrdinalIgnoreCase))
+            {
+                var traceIdRatio = ReadTraceIdRatio(configuration);
+                sampler = new TraceIdRatioBasedSampler(traceIdRatio);
+            }
+            else if (string.Equals(tracesSampler, "parentbased_always_on", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new ParentBasedSampler(new AlwaysOnSampler());
+            }
+            else if (string.Equals(tracesSampler, "parentbased_always_off", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new ParentBasedSampler(new AlwaysOffSampler());
+            }
+            else if (string.Equals(tracesSampler, "parentbased_traceidratio", StringComparison.OrdinalIgnoreCase))
+            {
+                var traceIdRatio = ReadTraceIdRatio(configuration);
+                sampler = new ParentBasedSampler(new TraceIdRatioBasedSampler(traceIdRatio));
+            }
+            else
+            {
+                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent($"Trace sampler configuration was found but the value '{tracesSampler}' is invalid and will be ignored.");
+            }
+        }
+
+        if (sampler != null)
+        {
+            OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent($"Trace sampler set to '{sampler.Description}' from configuration.");
+        }
```

Comment:
```suggestion
            if (sampler != null)
            {
                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent($"Trace sampler set to '{sampler.Description}' from configuration.");
            }
        }        
```

Moved that last log up into the block for if we find the config key. As it was written, if we had `stateSampler` without a config value, we would log that it came from config.